llvm/test/CodeGen/AMDGPU/wqm.ll

```diff
@@ -3361,7 +3361,7 @@ define amdgpu_ps void @test_for_deactivating_lanes_in_wave32(ptr addrspace(6) in
 ; GFX9-W64-NEXT:    s_buffer_load_dword s0, s[0:3], 0x0
 ; GFX9-W64-NEXT:    s_waitcnt lgkmcnt(0)
 ; GFX9-W64-NEXT:    v_cmp_le_f32_e64 vcc, s0, 0
-; GFX9-W64-NEXT:    s_andn2_b64 s[4:5], exec, vcc
```

Comment:
SI optimize exec mask pre-RA considers this as a unused exec copy and removes the copy (s[4:5] is not used).

Reply:
I don't understand - what copy is removed?

But anyway this looks like a change in behaviour. If the result of the s_andn2 is zero then we branch, so previously the exp instruction would run with exec!=0, but after your patch it runs with exec=0. Is that really intentional?